app/javascript/components/student/editor/TestSummary.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { useCallback } from 'react'
+import { TestStatus, Test } from './TestRunSummary'
+
+export function TestSummary({ test }: { test: Test }) {
+  const statusLabels = {
+    [TestStatus.PASS]: 'Passed',
+    [TestStatus.FAIL]: 'Failed',
+    [TestStatus.ERROR]: 'Failed',
+  }
+
+  return (
+    <details
+      open={test.status === TestStatus.FAIL || test.status === TestStatus.ERROR}
+    >
+      <summary>
+        {statusLabels[test.status]}: {test.name}
+      </summary>
+      {test.message && <p>Message: {test.message}</p>}
```

Comment:
This is dangerous.

If `message` is non null, but falsy, this will break.

Use explicit `expression ? <p>...</p> : null`